RustFxA/FxAWebViewModel.swift

```diff
@@ -261,6 +261,10 @@ extension FxAWebViewModel {
             NotificationManager().requestAuthorization { granted, error in
                 guard error == nil else { return }
                 if granted {
+                    self.profile.prefs.setBool(granted, forKey: PrefsKeys.Notifications.AllowAllNotifications)
```

Comment:
This could end up being problematic: If a user logs out and back in we'd overwrite their existing prefs. Also we ask for permission in onboarding as well and the prefs are not set there. I think we should not set the prefs after asking the user for permission and check in the settings if the pref is set and the user has permission. If the user has permission but no pref set the default value should be true.

Reply:
Very good catch. Setting `true` if the value is `nil` is a suitable fix?